database/postgres/seed.js

```diff
@@ -35,24 +36,61 @@ const createTables = () => new Promise(async (resolve, reject) => {
   .catch(err => console.log(err))
 
 // generate data and save to tables
-const seed = async (numOfRecords) => {
+const seed = async (numOfRecords = 1000, batchSize = 1000) => {
   try {
+    const timestamp = Date.now();
     await connect();
     await dropExistingTables();
     await createTables();
+    let itemsBuffer = [];
+    let descriptionBuffer = [];
+    let infoBuffer = [];
+    let configurationBuffer = [];
+    let productBuffer = [];
+    let counter = 0;
     for(let i = 1; i <= numOfRecords; i++) {
-      let { similarItemsQuery, descriptionQuery, infoQuery, configurationQuery, productQuery } = pgRecord(i);
-      await client.query(similarItemsQuery);
-      await client.query(descriptionQuery);
-      await client.query(infoQuery)
-      await client.query(configurationQuery);
-      await client.query(productQuery);
+      // get values from record converter
+      let { similarItemsValues, descriptionValues, productValues, infoValues, configurationValues } = pgRecord(i);
+
+      // push values up to buffers
+      itemsBuffer.push(similarItemsValues);
+      descriptionBuffer.push(descriptionValues);
+      infoBuffer.push(infoValues);
+      configurationBuffer.push(configurationValues);
+      productBuffer.push(productValues);
+
+      // when counter reaches batch size insert rows and clear buffers
+      counter++;
+      if (counter === batchSize || (i === numOfRecords && counter !== 0)) {
+        const items = format('INSERT INTO similarItems (id, similarItems) VALUES %L', itemsBuffer);
+        await client.query(items);
+        itemsBuffer = [];
+
+        const descriptions = format('INSERT INTO descriptions(id, itemDescription) VALUES %L', descriptionBuffer);
+        await client.query(descriptions);
+        descriptionBuffer = [];
+
+        const info = format('INSERT INTO info(id, itemColor, brand, isPrimeFreeOneDay, isFreeDelivery) VALUES %L', infoBuffer)
+        await client.query(info);
+        infoBuffer = [];
+
+        const configuration = format('INSERT INTO configurations(id, configuration) VALUES %L', configurationBuffer);
+        await client.query(configuration);
+        configurationBuffer = [];
+
+        const product = format('INSERT INTO products(id, itemName) VALUES %L', productBuffer)
+        await client.query(product);
+        productBuffer = [];
+
+        counter = 0;
+      }
     }
+    console.log((Date.now() - timestamp) / 1000)
     await disconnect();    
```

Comment:
this seeding function is so good and is so different than how I am going about this process. Is there a reason why you are using pg and pg-format instead of sequelize? Is is for performance reasons? Personal preference? Both?

Reply:
Was just using pg to try something new. I am not aware of any performance gains and so far cannot make an argument for using pg over sequelize. Thanks for compliments